fabric-mod-protocol-api-v1/src/main/java/net/fabricmc/fabric/impl/modprotocol/ModProtocolLocator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.modprotocol;
+
+import java.util.function.BiConsumer;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+import net.fabricmc.loader.api.metadata.CustomValue;
+import net.fabricmc.loader.api.metadata.ModMetadata;
+
+public class ModProtocolLocator {
+	public static void provide(BiConsumer<ModContainer, ModProtocolImpl> consumer) {
+		for (var mod : FabricLoader.getInstance().getAllMods()) {
+			create(mod, consumer);
+		}
+	}
+	private static void create(ModContainer container, BiConsumer<ModContainer, ModProtocolImpl> consumer) {
```

Comment:
Could this be implemented more fancy via `Codec::validate`? (`ModProtocolImpl.CODEC::validate`)

Reply:
Wrong line? This code parses it by hand, as you can't use codecs with loaders custom values (or at least I didn't know about it being an option). Also this is more flexible than that codec, which is only used for attaching mods to server's metadata

Reply:
Ah, yes, Fabric Loader doesn't give a JsonElement, I forgot about that